core/templates/dev/head/domain/state_card/StateCardObjectFactory.js

```diff
@@ -17,114 +17,189 @@
-  };
-
-  StateCard.prototype.setOppiaResponse = function(index, response) {
-    this._inputResponsePairs[index].oppiaResponse = response;
-  };
-
-  StateCard.prototype.setLastOppiaResponse = function(response) {
-    this.setOppiaResponse(this._inputResponsePairs.length - 1, response);
-  };
-
-  StateCard.prototype.setInteractionHtml = function(interactionHtml) {
-    this._interactionHtml = interactionHtml;
-  };
-
-  StateCard.prototype.setLeadsToConceptCard = function(leadsToConceptCard) {
-    this._leadsToConceptCard = leadsToConceptCard;
-  };
-
-  /**
-   * @param {string} stateName - The state name for the current card.
-   * @param {object} params - The set of parameters for the learner associated
-   *        with a card.
-   * @param {string} contentHtml - The HTML string for the content displayed on
-   *        the content card.
-   * @param {string} interactionHtml - The HTML that calls the interaction
-   *        directive for the current card.
-   * @param {Interaction} interaction - An interaction object that stores all
-   *        the properties of the card's interaction.
-   * @param {bool} leadsToConceptCard - Whether the current card leads to a
-   *        concept card in the exploration.
-   */
-  StateCard.createNewCard = function(
-      stateName, params, contentHtml, interactionHtml, interaction,
-      leadsToConceptCard) {
-    return new StateCard(
-      stateName, params, contentHtml, interactionHtml, interaction,
-      leadsToConceptCard, null, []);
-  };
-
-  return StateCard;
-}]);
+    };
+
+    StateCard.prototype.isCardTerminal = function() {
+      var interactionId = this.getInteractionId();
+      return (
+        interactionId && INTERACTION_SPECS[interactionId].is_terminal);
+    };
+
+    StateCard.prototype.getInteractionInstructions = function() {
+      var interactionId = this.getInteractionId();
+      return (
+        interactionId ? INTERACTION_SPECS[interactionId].instructions : '');
+    };
+
+    StateCard.prototype.getInteractionCustomizationArgs = function() {
+      var interaction = this.getInteraction();
+      if (!interaction) {
+        return null;
+      }
+      return interaction.customizationArgs;
+    };
+
+    StateCard.prototype.isInteractionInline = function() {
+      var interactionId = this.getInteractionId();
+      return (
+        !interactionId ||
+        INTERACTION_SPECS[interactionId].display_mode ===
+          INTERACTION_DISPLAY_MODE_INLINE);
+    };
+
+    StateCard.prototype.getCurrentParams = function() {
+      return this._currentParams;
+    };
+
+    StateCard.prototype.getContentHtml = function() {
+      return this._contentHtml;
+    };
+
+    StateCard.prototype.getInteractionHtml = function() {
+      return this._interactionHtml;
+    };
+
+    StateCard.prototype.getOppiaResponse = function(index) {
+      return this._inputResponsePairs[index].oppiaResponse;
+    };
+
+    StateCard.prototype.getLeadsToConceptCard = function() {
```

Comment:
Is this relevant to what's displayed? If not it doesn't belong here.

Reply:
This also serves a similar purpose to getDestStateName and is used as a condition for some UI elements.

Reply:
OK, see above comments for getDestStateName.

Reply:
Done, removed this.